modules/tinymce/src/core/main/ts/paste/Clipboard.ts

```diff
@@ -30,9 +31,10 @@ const uniqueId = PasteUtils.createIdGenerator('mceclip');
 
 const doPaste = (editor: Editor, content: string, internal: boolean, pasteAsText: boolean): void => {
   const args = ProcessFilters.process(editor, content, internal);
-
   if (!args.cancelled) {
-    SmartPaste.insertContent(editor, args.content, pasteAsText);
+    const content = args.content;
+    SmartPaste.insertContent(editor, content, pasteAsText);
+    fireInputEvent(editor, 'insertFromPaste', { data: content });
```

Comment:
I don't think this event is correct. Check this codepen in Chrome https://codepen.io/spocke78/pen/PoydNMO it only populates the dataTransfer and only on beforeInput not on input. Also data never seems to be populated at least not on Chrome. So I think we should just mimic the browser here and not include the data. If people want to intercept paste data and alter it there are other ways. Like `BeforeSetContent` having a `paste` flag when stuff is coming from the clipboard.

Reply:
@spocke Thanks for the demo! I see now that `data` is null across all browsers. 

However, both Firefox and Safari seems to have a populated dataTransfer property in the `input` event as well, but I'm happy to go with Chrome's approach of having both `data` and `dataTransfer` properties being `null`. 

Having said that, is it be worth also dispatching a `beforeInput` event as well which has a `DataTransfer` object containing the data, to further mimic browser behavior. If so, I will split that into a separate ticket.